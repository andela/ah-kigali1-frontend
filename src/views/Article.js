import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import addArticle from "../redux/actions/index";
// import "../index.css";

const mapDispatchToProps = dispatch => ({
  addArticle: () => dispatch(addArticle())
});

const FetchedArticles = props => (
  <div>
    <h1>Fetch some articles from dummy app</h1>

    <button
      // eslint-disable-next-line react/destructuring-assignment
      onClick={() => props.addArticle()}
      className="btn"
      type="submit"
    >
      Fetch more articles
    </button>
  </div>
);

FetchedArticles.propTypes = {
  // eslint-disable-next-line react/require-default-props
  addArticle: PropTypes.func
};

const Articles = connect(
  null,
  mapDispatchToProps
)(FetchedArticles);

export default Articles;
