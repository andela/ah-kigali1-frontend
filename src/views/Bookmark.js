import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { withRouter } from "react-router-dom";
import BookmarkComponent from "../components/common/Bookmarks";
import bookmarkIcon from "../assets/img/bookmark-icons.svg";
import bookmarkedIcon from "../assets/img/bookmarked.svg";
import { handleBookmark } from "../redux/actions/bookmarkActions";

export class Bookmark extends Component {
  static chooseBookmarkIcon(condition) {
    return condition ? bookmarkedIcon : bookmarkIcon;
  }

  render() {
    const {
      bookmark,
      isBookmarked,
      slug,
      isSubmitting,
      bookmarkedArticles,
      location,
      history
    } = this.props;
    if (!Object.prototype.hasOwnProperty.call(bookmarkedArticles, slug)) {
      Object.assign(bookmarkedArticles, { [slug]: isBookmarked });
    }
    const theBookmarkIcon = Bookmark.chooseBookmarkIcon(
      bookmarkedArticles[slug]
    );
    return (
      <BookmarkComponent
        onClick={() =>
          bookmark(slug, bookmarkedArticles[slug], { location, history })
        }
        icon={theBookmarkIcon}
        disabled={isSubmitting}
      />
    );
  }
}

export const mapStateToProps = state => ({
  ...state.bookmark
});

export const mapDispatchToProps = dispatch => ({
  bookmark: (slug, isBookmarked, { history, location }) =>
    dispatch(handleBookmark(slug, isBookmarked, { history, location }))
});

Bookmark.propTypes = {
  bookmark: PropTypes.func.isRequired,
  isBookmarked: PropTypes.bool.isRequired,
  bookmarkedArticles: PropTypes.shape({}).isRequired,
  slug: PropTypes.string.isRequired,
  isSubmitting: PropTypes.bool.isRequired,
  location: PropTypes.shape([]).isRequired,
  history: PropTypes.shape({}).isRequired
};

export default withRouter(
  connect(
    mapStateToProps,
    mapDispatchToProps
  )(Bookmark)
);
