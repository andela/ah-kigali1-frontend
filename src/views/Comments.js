/* eslint-disable jsx-a11y/click-events-have-key-events */
/* eslint-disable no-shadow */
import React, { Component } from "react";
import Textarea from "react-textarea-autosize";
import PropTypes from "prop-types";
import moment from "moment";
import authorImage from "../assets/img/user.jpg";
import Button from "../components/common/Buttons/BasicButton";
import DeleteDialogue from "../components/common/Dialog/DeleteDialogue";
import heartIcon from "../assets/img/heart.svg";

export class Comments extends Component {
  state = {
    editMode: false,
    index: null
  };

  onEnterPress = (e, id) => {
    const { editComment } = this.props;
    if (e.keyCode === 13 && e.shiftKey === false) {
      e.preventDefault();
      editComment(id);
      this.setState({ editMode: false });
    }
  };

  isEditing = (body, index) => {
    const { onSetBodyEdit } = this.props;
    this.setState({ editMode: true, index });
    onSetBodyEdit({ value: body });
  };

  handleComment = id => {
    const { editComment } = this.props;
    editComment(id);
    this.setState({ editMode: false });
  };

  render() {
    const { editMode, index: commentIndex } = this.state;
    const {
      comments,
      updatedBody,
      deleteComment,
      handleCommentsInputEdit
    } = this.props;

    return Object.values(comments).length
      ? Object.values(comments).map((comment, index) => {
          const { body, like, id, createdAt, author } = comment;
          const { image, lastName, firstName, username } = author;
          return (
            <div className="article-comments--existing__desktop" key={id}>
              <div className="avatar-wrapper comment-avatar-wrapper">
                <img
                  src={image || authorImage}
                  alt="Avatar"
                  className="avatar"
                />
                <span className="comment-author_name ">
                  {username && firstName && lastName
                    ? `${firstName} ${lastName.charAt(0) ||
                        username.charAt(0)}.`
                    : username}
                </span>
              </div>
              <div className="article-comments--existing-text">
                {editMode && Number(commentIndex) === index ? (
                  <div>
                    <Textarea
                      id="edit-comment-textarea"
                      type="textarea"
                      onChange={handleCommentsInputEdit}
                      onKeyDown={e => this.onEnterPress(e, id)}
                      name="bodyEdit"
                      value={updatedBody}
                    />
                    <Button
                      id="update-comment"
                      className="btn delete_article"
                      onClick={() => this.handleComment(id)}
                      title="Update"
                    />
                    <Button
                      id="close-update-comment"
                      className="btn delete_article"
                      onClick={() => this.setState({ editMode: false })}
                      title="Cancel"
                    />
                  </div>
                ) : (
                  <div>
                    <div className="time-ago">
                      Posted {moment(createdAt).fromNow()}
                    </div>
                    <div className="comment-body" key={id}>
                      {body}
                    </div>
                    <div className="comment-edit-delete">
                      <DeleteDialogue
                        id="delete-comment"
                        deleteComment={() => deleteComment(id)}
                      />
                      <div
                        className="edit-comment"
                        onClick={() => this.isEditing(body, index)}
                      >
                        <i className="fa fa-edit" />
                      </div>
                    </div>
                    <div className="article-comments--actions">
                      <span className="comment-like">
                        <div className="icons">
                          <img src={heartIcon} alt="likes" className="likes" />
                          <div>{like}</div>
                        </div>
                      </span>
                    </div>
                  </div>
                )}
              </div>
            </div>
          );
        })
      : "";
  }
}

Comments.propTypes = {
  comments: PropTypes.shape([]).isRequired,
  updatedBody: PropTypes.string.isRequired,
  onSetBodyEdit: PropTypes.func.isRequired,
  deleteComment: PropTypes.func.isRequired,
  editComment: PropTypes.func.isRequired,
  handleCommentsInputEdit: PropTypes.func.isRequired
};

export default Comments;
