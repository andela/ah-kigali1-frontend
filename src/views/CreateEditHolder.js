import React, { Component } from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import RichTextEditor from "react-rte";
import Validator from "../utils/validator";
import { checkNextProps } from "../utils/helperFunctions";
import Input from "../components/common/Inputs/TextInput";
import Button from "../components/common/Buttons/BasicButton";
import Error from "../components/common/Message/error";
import Tag from "../components/common/Tags/NewTag";
import { toolbarConfig } from "../utils/editor/editorConfig";
import {
  editArticle,
  newArticle,
  fetchOneArticle
} from "../redux/actions/newArticle";

export const mapDispatchToProps = dispatch => ({
  fetchArticleToEdit: slug => dispatch(fetchOneArticle(slug)),
  createArticle: (article, history) => dispatch(newArticle(article, history)),
  editOneArticle: (article, slug, history) =>
    dispatch(editArticle(article, slug, history))
});

export const mapStateToProps = state => ({
  ...state.newArticle,
  ...state.auth
});

export class Article extends Component {
  constructor(props) {
    super(props);
    this.state = {
      edit: false,
      tag: "",
      title: "",
      isFetching: false,
      description: "",
      tagsList: [],
      error: "",
      value: RichTextEditor.createEmptyValue()
    };
  }

  componentDidMount = () => {
    const {
      match: {
        params: { slug }
      },
      fetchArticleToEdit,
      history: {
        location: { pathname }
      }
    } = this.props;
    if (pathname !== "/articles/new") {
      this.setState({ edit: true, isFetching: true });
      fetchArticleToEdit(slug);
    }
  };

  componentWillReceiveProps = nextProps => {
    const { articleError } = nextProps;
    if (articleError) {
      this.setState({ error: articleError });
      return setTimeout(() => this.setState({ error: "" }), 3000);
    }
    const { article } = this.props;

    if (checkNextProps(nextProps.article, article)) {
      this.setState({
        ...nextProps.article,
        isFetching: false,
        value: RichTextEditor.createValueFromString(
          nextProps.article.body,
          "html"
        )
      });
    }
  };

  onEditorChange = value => {
    this.setState({ value });
  };

  onRemoveTag = value => {
    const { tagsList } = this.state;
    const index = tagsList.indexOf(value);
    tagsList.splice(index, 1);
    this.setState({ tagsList });
  };

  cancelEdit = () => {
    const { history } = this.props;
    return history.push("/");
  };

  handleInputChange = event => {
    const { name, value } = event.target;
    this.setState({ [name]: value });
  };

  submitArticle = () => {
    const {
      editOneArticle,
      createArticle,
      history,
      history: {
        location: { pathname }
      },
      match: {
        params: { slug }
      }
    } = this.props;
    const { title, value, description } = this.state;

    const error = Validator.newArticleValidation({
      title,
      body: value.toString("html"),
      description
    });
    if (error) {
      this.setState({ error });
      return this.clearErrorMessage();
    }
    this.setState({
      error: ""
    });
    if (pathname !== "/articles/new") {
      return editOneArticle(
        {
          title,
          body: value.toString("html"),
          description
        },
        slug,
        history
      );
    }
    return createArticle(
      {
        title,
        body: value.toString("html"),
        description
      },
      history
    );
  };

  handleNewTag = event => {
    const { tag, tagsList } = this.state;
    if (event.keyCode === 13 && tag.length) {
      this.setState({ tag: "", tagsList: [...tagsList, tag] });
    }
  };

  clearErrorMessage = () =>
    setTimeout(() => this.setState({ error: "" }), 3000);

  displayErrorMessage = () => {
    const { error } = this.state;
    return (
      <div className="error error__message">
        {error.length ? <Error title={error} /> : ""}
      </div>
    );
  };

  render() {
    const {
      value,
      edit,
      tag,
      title,
      body,
      description,
      tagsList,
      isFetching
    } = this.state;

    const { isSubmitting } = this.props;
    return isFetching ? (
      ""
    ) : (
      <div>
        {this.displayErrorMessage()}
        <div className="new-article-container container">
          <div className="buttons">
            {edit ? (
              <Button
                className="button--dark cancel"
                title="Cancel"
                disabled={isSubmitting}
                onClick={this.cancelEdit}
              />
            ) : (
              ""
            )}
            <Button
              className="button--dark publish"
              title={edit ? "Save" : "Publish"}
              disabled={isSubmitting}
              onClick={this.submitArticle}
            />
          </div>
          <div className="main-article">
            <div className="title">
              <Input
                className="title__input"
                type="text"
                value={title}
                placeholder="The title should come here"
                name="title"
                onChange={e => this.handleInputChange(e)}
              />
            </div>
            <hr className="line__title" />
            <div className="description">
              <Input
                className="description__input"
                type="text"
                value={description}
                placeholder="Description should be here"
                name="description"
                onChange={e => this.handleInputChange(e)}
              />
            </div>
            <hr className="line__title" />
            <div className="tags">
              {tagsList.map(newTag => (
                <Tag
                  title={newTag}
                  key={newTag}
                  onClick={() => this.onRemoveTag(newTag)}
                />
              ))}
              <Input
                type="text"
                id="new-tag"
                placeholder="new tag"
                name="tag"
                className="tag"
                value={tag}
                onKeyDown={e => this.handleNewTag(e)}
                onChange={e => this.handleInputChange(e)}
              />
            </div>
            <RichTextEditor
              editorClassName="demo_editor"
              placeholder={body}
              value={value}
              onChange={e => this.onEditorChange(e)}
              toolbarConfig={toolbarConfig}
            />
          </div>
        </div>
      </div>
    );
  }
}

Article.propTypes = {
  editOneArticle: PropTypes.func.isRequired,
  createArticle: PropTypes.func.isRequired,
  fetchArticleToEdit: PropTypes.func.isRequired,
  articleError: PropTypes.shape({}).isRequired,
  isSubmitting: PropTypes.bool,
  article: PropTypes.shape({
    title: PropTypes.string.isRequired,
    body: PropTypes.string.isRequired,
    message: PropTypes.string,
    response: PropTypes.object,
    description: PropTypes.string,
    tagsList: PropTypes.arrayOf(PropTypes.string)
  }).isRequired,
  match: PropTypes.shape({
    params: PropTypes.shape({
      slug: PropTypes.string.isRequired
    }).isRequired
  }).isRequired,
  history: PropTypes.shape({
    push: PropTypes.func.isRequired
  }).isRequired
};

Article.defaultProps = {
  isSubmitting: false
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Article);
