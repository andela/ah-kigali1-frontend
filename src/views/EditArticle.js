import React, { Component } from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";
import RichTextEditor from "react-rte";
import Navbar from "../components/common/AppBars/Navbar";
import Validator from "../utils/validator";
import Input from "../components/common/Inputs/TextInput";
import Button from "../components/common/Buttons/BasicButton";
import Success from "../components/common/Message/success";
import Error from "../components/common/Message/error";
import Tag from "../components/common/Tags/NewTag";
import { toolbarConfig } from "../utils/editor/editorConfig";
import {
  handleInputField,
  editArticle,
  handleCreateTag,
  fetchOneArticle,
  removeTag
} from "../redux/actions/newArticle";

export const mapDispatchToProps = dispatch => ({
  onInputChange: field => dispatch(handleInputField(field)),
  fetchArticleToEdit: slug => dispatch(fetchOneArticle(slug)),
  newTag: tag => dispatch(handleCreateTag(tag)),
  editOneArticle: (article, slug) => dispatch(editArticle(article, slug)),
  handleRemoveTag: tag => dispatch(removeTag(tag))
});

export const mapStateToProps = (state, ownProps) => ({
  response: state.article.response,
  article: state.article,
  ...ownProps
});

export class EditArticle extends Component {
  constructor(props) {
    super(props);

    this.state = {
      tag: "",
      title: props.article.title,
      isFetching: false,
      description: props.article.description,
      tagsList: props.article.tagsList,
      error: "",
      value: RichTextEditor.createValueFromString(props.article.body, "html")
    };
  }

  componentDidMount = () => {
    const { match, fetchArticleToEdit } = this.props;
    const { slug } = match.params;
    fetchArticleToEdit(slug);
  };

  componentWillReceiveProps = nextProps => {
    if (nextProps.article.article_error) {
      const { history } = this.props;
      this.setState({ error: nextProps.article.article_error });
      return history.push("/not_found");
    }
    const { article } = this.props;
    const { title, description, tagsList, body } = article;
    if (
      nextProps.article.title !== title ||
      nextProps.article.description !== description ||
      nextProps.article.tagsList !== tagsList ||
      nextProps.article.body !== body
    ) {
      this.setState({
        title: nextProps.article.title,
        tagsList: nextProps.article.tagsList,
        isFetching: false,
        description: nextProps.article.description,
        value: RichTextEditor.createValueFromString(
          nextProps.article.body,
          "html"
        )
      });
    }
  };

  onEditorChange = value => {
    this.setState({ value });
    const { onInputChange } = this.props;
    onInputChange({ name: "body", value: value.toString("html") });
  };

  onTagInputChange = event => {
    this.setState({
      tag: event.target.value
    });
  };

  onRemoveTag = value => {
    const { handleRemoveTag } = this.props;
    handleRemoveTag(value);
  };

  cancelEdit = () => {
    const { history } = this.props;
    return history.push("/");
  };

  handleInputChange = event => {
    const { name, value } = event.target;
    const { onInputChange } = this.props;
    this.setState({ [name]: value });
    onInputChange({ name, value });
  };

  submitArticle = () => {
    const { article, editOneArticle, match } = this.props;
    const { title, body, description, tagsList } = article;
    const { slug } = match.params;
    const error = Validator.newArticleValidation(article);
    if (error && error.length) {
      return this.setState({ error });
    }
    this.setState({
      error: ""
    });
    editOneArticle({ title, body, description, tagsList }, slug);
  };

  handleNewTag = event => {
    const { tag } = this.state;
    if (event.keyCode === 13 && tag.length) {
      const { newTag } = this.props;
      newTag({ tag });
      this.setState({ tag: "" });
    }
  };

  render() {
    const {
      error,
      value,
      tag,
      title,
      body,
      description,
      tagsList,
      isFetching
    } = this.state;
    const { isSubmitting, response } = this.props;
    return isFetching ? (
      false
    ) : (
      <div>
        <Navbar />

        <div className="new-article-container container">
          <div className="buttons">
            <Button
              className="button--dark cancel"
              title="Cancel"
              disabled={isSubmitting}
              onClick={() => this.cancelEdit()}
            />
            <Button
              className="button--dark publish"
              title="Save"
              disabled={isSubmitting}
              onClick={e => this.submitArticle(e)}
            />
          </div>
          <div className="error">
            {error.length ? <Error title={error} /> : false}
          </div>
          <div className="success message">
            {response ? <Success title={response.message} /> : false}
          </div>
          <div className="main-article">
            <div className="title">
              <Input
                className="title__input"
                type="text"
                value={title}
                placeholder="The title should come here"
                name="title"
                onChange={e => this.handleInputChange(e)}
              />
            </div>
            <hr className="line__title" />
            <div className="description">
              <Input
                className="description__input"
                type="text"
                value={description}
                placeholder="Description should be here"
                name="description"
                onChange={e => this.handleInputChange(e)}
              />
            </div>
            <hr className="line__title" />
            <div className="tags">
              {tagsList.map(newTag => (
                <Tag
                  title={newTag}
                  key={newTag}
                  onClick={() => this.onRemoveTag(newTag)}
                />
              ))}
              <Input
                type="text"
                id="new-tag"
                placeholder="new tag"
                name="tag"
                className="tag"
                value={tag}
                onKeyUp={e => this.handleNewTag(e)}
                onChange={e => this.onTagInputChange(e)}
              />
            </div>
            <RichTextEditor
              editorClassName="demo_editor"
              placeholder={body}
              value={value}
              onChange={e => this.onEditorChange(e)}
              toolbarConfig={toolbarConfig}
            />
          </div>
        </div>
      </div>
    );
  }
}

EditArticle.propTypes = {
  onInputChange: PropTypes.func.isRequired,
  editOneArticle: PropTypes.func.isRequired,
  newTag: PropTypes.func.isRequired,
  handleRemoveTag: PropTypes.func.isRequired,
  fetchArticleToEdit: PropTypes.func.isRequired,
  isSubmitting: PropTypes.bool,
  response: PropTypes.string,
  article: PropTypes.shape({
    title: PropTypes.string.isRequired,
    body: PropTypes.string.isRequired,
    article_error: PropTypes.string,
    description: PropTypes.string,
    tagsList: PropTypes.arrayOf(PropTypes.string)
  }).isRequired,
  match: PropTypes.shape({
    params: PropTypes.shape({
      slug: PropTypes.string.isRequired
    }).isRequired
  }).isRequired,
  history: PropTypes.shape({
    push: PropTypes.func.isRequired
  }).isRequired
};

EditArticle.defaultProps = {
  isSubmitting: false,
  response: false
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(EditArticle);
