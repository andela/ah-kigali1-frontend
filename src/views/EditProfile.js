import React, { Component } from "react";
import { connect } from "react-redux";
import { fieldRemover } from "../helpers/helpers";
import Input from "../components/common/Inputs/input";
import Confirm from "../components/common/Buttons/confirm";
import {
  fetchCurrentUser,
  handleFormInput,
  saveUpdatedUser,
  uploadImage
} from "../redux/actions";
import ErrorMessage from "../components/common/Message/error";
import SuccessMessage from "../components/common/Message/success";


const usernameFromLocalStorage = "username";

export class EditProfile extends Component {
  componentDidMount() {
    this.props.fetchCurrentUser(usernameFromLocalStorage);
  }
  handleChange = e => {
    this.props.onInputChange({ field: e.target.name, value: e.target.value });
  };
  handleImageChange = e => {
    this.props.uploadImage(e.target.files[0]).then(() => {
      const { loggedInUser } = this.props;
      this.props.saveUpdatedUser(
        fieldRemover(loggedInUser),
        usernameFromLocalStorage
      );
    });
  };
  handleSubmit = e => {
    const { loggedInUser } = this.props;
    this.props.saveUpdatedUser(
      fieldRemover(loggedInUser),
      usernameFromLocalStorage
    );
  };

  render() {
    const { error, message } = this.props;
    const {
      username,
      email,
      firstName,
      lastName,
      bio,
      phone,
      address,
      image
    } = this.props.loggedInUser;
    return (
      <div className="main update-profile-container">
        <div className="update">
          <div className="update-left">
            <div className="logo-container">
              <img
                src={require("../assets/img/main-logo.svg")}
                alt="logo"
                className="logo"
              />
              <h2>Author's Heaven</h2>
            </div>
            <div className="avatar-container">
              <img
                src={image}
                className="avatar"
                alt="profilePicture"
                name="image"
              />
              <input
                className="profile_picture"
                name="image"
                onChange={this.handleImageChange}
                type="file"
                alt="profilePicture"
                style={{ display: "block", cursor: "pointer" }}
              />
            </div>
          </div>

          <div className="update-right">
            <div className="update-form">
              <p>Edit profile</p>
              <div>{error ? <ErrorMessage title={error} /> : ""}</div>
              <div>{message ? <SuccessMessage title={message} /> : ""}</div>
              <Input
                className="input_field"
                name="firstName"
                type="text"
                placeholder="First name"
                onChange={this.handleChange}
                value={firstName}
              />
              <Input
                name="lastName"
                type="text"
                placeholder="Last name"
                onChange={this.handleChange}
                value={lastName}
              />
              <textarea
                rows="7"
                cols="53"
                type="textarea"
                name="bio"
                placeholder="Bio"
                onChange={this.handleChange}
                value={bio}
              />
              <Input
                name="email"
                type="email"
                placeholder="Email"
                onChange={this.handleChange}
                value={email}
                disabled
              />
              <Input
                name="phone"
                type="tel"
                placeholder="Phone"
                onChange={this.handleChange}
                value={phone}
              />
              <Input
                name="address"
                type="text"
                placeholder="City"
                onChange={this.handleChange}
                value={address}
              />
              <Input
                name="username"
                type="text"
                placeholder="Username"
                onChange={this.handleChange}
                value={username}
              />
              <Confirm onClick={this.handleSubmit} title="Save" />
            </div>
          </div>
        </div>
      </div>
    );
  }
}
export const mapStateToProps = state => ({
  loggedInUser: state.user.profile,
  loading: state.user.loading,
  error: state.user.error,
  message: state.user.message
});

export const mapDispatchToProps = dispatch => {
  return {
    fetchCurrentUser: username => dispatch(fetchCurrentUser(username)),
    onInputChange: payload => dispatch(handleFormInput(payload)),
    saveUpdatedUser: (data, username) =>
      dispatch(saveUpdatedUser(data, username)),
    uploadImage: file => dispatch(uploadImage(file))
  };
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(EditProfile);
