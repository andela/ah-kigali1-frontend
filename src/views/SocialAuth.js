import React, { Component } from "react";
import PropTypes from "prop-types";
import { Redirect } from "react-router-dom";
import { connect } from "react-redux";
import { handleUserLogin } from "../redux/actions/socialAuthCations";
import { parseURL } from "../utils/helperFunctions";

class SocialAuth extends Component {
  componentWillMount() {
    const {
      location: { search },
      handleUserLogin: loginUser
    } = this.props;
    const token = parseURL("?token=", search);
    loginUser(token);
  }

  render() {
    const { isSubmitting, socialAuthFailed, socialAuthSuccess } = this.props;
    if (isSubmitting) {
      return (
        <div>
          <h3>Loading.......</h3>
        </div>
      );
    }
    if (socialAuthFailed) {
      return <Redirect to="/sign_In" />;
    }
    if (socialAuthSuccess) {
      return <Redirect to="/" />;
    }
  }
}
SocialAuth.propTypes = {
  location: PropTypes.object.isRequired,
  handleUserLogin: PropTypes.func.isRequired,
  isSubmitting: PropTypes.bool.isRequired,
  socialAuthSuccess: PropTypes.bool.isRequired,
  socialAuthFailed: PropTypes.bool.isRequired
};
const mapStateToProps = state => {
  const { socialAuth } = state;
  return {
    ...socialAuth
  };
};
export default connect(
  mapStateToProps,
  { handleUserLogin }
)(SocialAuth);
