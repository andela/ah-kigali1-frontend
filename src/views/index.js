import React from "react";
import { BrowserRouter as Router, Route, Switch } from "react-router-dom";
import Login from "./Login";
import ResetPassword from "./ResetPassword";
import Home from "./Home";
import UpdatePassword from "./UpdatePassword";
import SocialAuth from "./SocialAuth";
import NewArticle from "./NewArticle";
import EditArticle from "./EditArticle";
import NotFound from "./NotFound";
import ReadArticle from "./ReadArticle";

const Routers = () => (
  <Router>
    <Switch>
      <Route exact path="/" component={Home} />
      <Route exact path="/sign_in" component={Login} />
      <Route exact path="/social_auth" component={SocialAuth} />
      <Route exact path="/reset_password" component={ResetPassword} />
      <Route exact path="/update_password" component={UpdatePassword} />
      <Route exact path="/articles/:slug/edit" component={EditArticle} />
      <Route exact path="/articles/new_article" component={NewArticle} />
      <Route exact path="/not_found" component={NotFound} />
      <Route exact path="/articles/:slug" component={ReadArticle} />
    </Switch>
  </Router>
);
export default Routers;
