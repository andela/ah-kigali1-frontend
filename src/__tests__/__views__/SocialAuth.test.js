import React from "react";
import toJson from "enzyme-to-json";
import { shallow } from "enzyme";
import { SocialAuth } from "../../views/SocialAuth";

const [handleUserLogin] = new Array(2).fill(jest.fn());
const props = {
  location: { search: "?token=23qwertyuhjbgty" },
  handleUserLogin,
  isSubmitting: true,
  socialAuthSuccess: false,
  socialAuthFailed: false
};
const wrapper = shallow(<SocialAuth {...props} />);

describe("Social auth component", () => {
  let instance;
  beforeEach(() => {
    instance = wrapper.instance();
    jest.spyOn(instance, "handleFailure");
    jest.spyOn(instance, "handleSuccess");
  });
  afterEach(() => {
    handleUserLogin.mockClear();
    instance.handleFailure.mockClear();
    instance.handleSuccess.mockClear();
    wrapper.setProps({
      ...props
    });
  });
  it("should match the snapshot", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
    expect(handleUserLogin).toHaveBeenCalledWith("23qwertyuhjbgty");
  });
  it("return failure and redirect to sign_in page", () => {
    wrapper.setProps({
      isSubmitting: false,
      socialAuthFailed: true
    });
    expect(instance.handleSuccess.mock.calls.length).toBe(0);
    expect(instance.handleFailure.mock.calls.length).toBe(1);
  });
  it("return failure and redirect to sign_in page", () => {
    wrapper.setProps({
      isSubmitting: false,
      socialAuthSuccess: true
    });
    expect(instance.handleSuccess.mock.calls.length).toBe(1);
    expect(instance.handleFailure.mock.calls.length).toBe(0);
  });
});
