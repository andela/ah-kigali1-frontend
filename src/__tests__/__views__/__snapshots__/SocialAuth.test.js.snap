// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Social auth component should match the snapshot 1`] = `
<div>
  <h3>
    Loading.......
  </h3>
</div>
`;
